crates/ruff/src/rules/pylint/rules/continue_in_finally.rs

```diff
@@ -0,0 +1,72 @@
+use rustpython_parser::ast::{Stmt, StmtKind};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::types::Range;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `continue` inside of a `finally` clause
+///
+/// ## Why is this bad?
+/// `continue` is not supported inside a `finally` clause; this can cause a SyntaxError
+///
+///
+/// ## Example
+/// ```python
+/// while True:
+///  try:
+///     pass
+///  finally:
+///     continue
+/// ```
+///
+/// Use instead:
+/// ```python
+/// while True:
+///  try:
+///     pass
+///  except Exception:
+///     pass
+///  else:
+///     continue
+/// ```
+#[violation]
+pub struct ContinueInFinally;
+
+impl Violation for ContinueInFinally {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`continue` not supported inside a `finally` clause")
+    }
+}
+
+fn traverse_body(checker: &mut Checker, body: &[Stmt]) {
+    for stmt in body {
+        if matches!(stmt.node, StmtKind::Continue { .. }) {
+            checker
+                .diagnostics
+                .push(Diagnostic::new(ContinueInFinally, Range::from(stmt)));
+        }
+
+        match &stmt.node {
+            StmtKind::If { body, .. }
+            | StmtKind::With { body, .. }
+            | StmtKind::AsyncWith { body, .. } => {
+                traverse_body(checker, body);
```

Comment:
search for `Vec<Stmt<U>>` in `StmtKind` source found some other cases to traverse-

`orelse`  in a couple of nodes, including `If`, `For`, `AsyncFor` and `While`

Reply:
I always forget those `for..else` and `while..else` statements exist 🫠